analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -897,6 +897,27 @@ public void CatchWhen_Finally()
             validator.TagStates("End").Should().HaveCount(1).And.ContainSingle(x => HasNoException(x));
         }
 
+        [TestMethod]
+        public void Finally_NestedLambda_ShouldNotFail()
+        {
+            const string code = @"
+Tag(""Unreachable - outer CFG is not analyzed"");
+try
+{
+    Action a = () =>
+    {
+        var tag = ""Before"";
+        Tag(""CanThrow"");
+        tag = ""After"";
+    };
+}
+finally
+{
+    var nonEmptyFinallyBlock = true;
+}";
+            SETestContext.CreateCSLambda(code, "tag").Validator.ValidateTagOrder("Before", "CanThrow", "After");
```

Comment:
"tag" is used for different purposes and is highly confusing in this context. It also seems very dangerous to use it for identifying the lambda. What about `// lambda marker` instead?